components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java

```diff
@@ -0,0 +1,358 @@
+            throws Exception {
+        when(scimUserOperationListener.isEnable()).thenReturn(isEnabled);
+        when(userStoreManager.isSCIMEnabled()).thenReturn(isSCIMEnabled);
+
+        assertTrue(scimUserOperationListener.doPreSetUserClaimValues(userName, claims, profile, userStoreManager));
+    }
+
+    @Test
+    public void testDoPostSetUserClaimValues() throws Exception {
+        assertTrue(scimUserOperationListener.doPostSetUserClaimValues(userName, claims, profile, userStoreManager));
+    }
+
+    @Test
+    public void testDoPreDeleteUserClaimValues() throws Exception {
+        assertTrue(scimUserOperationListener.doPreDeleteUserClaimValues(userName, roleList, userName,
+                userStoreManager));
+    }
+
+    @Test
+    public void testDoPostDeleteUserClaimValues() throws Exception {
+        assertTrue(scimUserOperationListener.doPostDeleteUserClaimValues(userName, userStoreManager));
+    }
+
+    @Test
+    public void testDoPreDeleteUserClaimValue() throws Exception {
+        assertTrue(scimUserOperationListener.doPreDeleteUserClaimValue(anyString(), anyString(), anyString(),
+                eq(userStoreManager)));
+    }
+
+    @Test
+    public void testDoPostDeleteUserClaimValue() throws Exception {
+        assertTrue(scimUserOperationListener.doPostDeleteUserClaimValue(userName, userStoreManager));
+    }
+
+    @Test
+    public void testDoPreAddRole() throws Exception {
+        assertTrue(scimUserOperationListener.doPreAddRole(userName, roleList, permissions, userStoreManager));
+
+    }
+
+    @DataProvider(name = "testDoPostAddRoleData")
+    public Object[][] testDoPostAddRoleData() {
+
+        return new Object[][]{
+                {true, true, true, "testDomain"},
+                {true, true, false, null},
+                {false, false, true, "testDomain"},
+                {true, false, false, "testDomain"},
+                {false, true, true, "testDomain"}
+        };
+    }
+
+    @Test(dataProvider = "testDoPostAddRoleData")
+    public void testDoPostAddRole(boolean isEnabled, boolean isSCIMEnabled, boolean isGroupExisting, String domainName)
+            throws Exception {
+        when(scimUserOperationListener.isEnable()).thenReturn(isEnabled);
+        when(userStoreManager.isSCIMEnabled()).thenReturn(isSCIMEnabled);
+        when(groupDAO.isExistingGroup(anyString(), anyInt())).thenReturn(isGroupExisting);
+        whenNew(GroupDAO.class).withNoArguments().thenReturn(groupDAO);
+        when(UserCoreUtil.getDomainName((RealmConfiguration) anyObject())).thenReturn(domainName);
+        when(UserCoreUtil.addDomainToName(anyString(), anyString())).thenReturn("testRoleNameWithDomain");
+        when(SCIMCommonUtils.getGroupNameWithDomain(anyString())).thenReturn("testRoleNameWithDomain");
+
+        assertTrue(scimUserOperationListener.doPostAddRole(roleName, userList, permissions, userStoreManager));
+    }
+
```

Comment:
If possible integrate this two methods(testDoPostAddRole,testDoPreDeleteRole) in to one or define a common method and use that

Reply:
Updated the PR.